### PR TITLE
Bug fixing for printf return value

### PR DESCRIPTION
**Pull request by ChenfengWei0**
Related issue: #1089 

example
```c++
char *data = "123456";
int x = printf("%s", data);
assert(x == 6); // should hold
```

- However, the return value is always zero in Windows, probably due to the different symbol names in Windows (e.g. `__stdio_common_vfprintf`). Also, the error log shows a function body missing:

```sh
no body for function __acrt_iob_func
```
- Related info: [The printf and scanf family of functions are now defined inline - Microsoft](https://learn.microsoft.com/en-us/cpp/porting/visual-cpp-change-history-2003-2015?redirectedfrom=MSDN&view=msvc-170#BK_CRT)

may be fixed in another PR

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]
Thanks for submitting this PR, @ChenfengWei0.

**Review by rafaelsamenezes — APPROVED** [review]

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @ChenfengWei0.

File: src/util/array2string.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  if(src.type.id() != irept::id_array)
+    return;
+
+  std::string value_str;
+  size_t cnt = 0;
+  forall_operands(it, src.value)
+  {
+    std::string op_str = it->cformat().as_string();
+    if(cnt < src.value.operands().size() - 1)
+      value_str.push_back(op_str[1]);
```
Comment (kunjsong01) [inline src/util/array2string.cpp:14]: Would it be possible to add a comment here explaining why "op_str[1]"?

File: regression/cstd/printf_bug_1/main.c [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include <stdio.h>
+#include <assert.h>
+int main()
+{
+	char *s = "abcde123415";
```
Comment (kunjsong01) [inline regression/cstd/printf_bug_1/main.c:5]: indentation formatting

**Review by kunjsong01 — COMMENTED** [review]
just some nitpicks

File: src/util/array2string.cpp (hunk 1 above)
Comment (ChenfengWei0) [reply src/util/array2string.cpp:14]: Sure. I will add some comments. Basically it's the reversal of the `string2array()`:
```c++
    if(ch >= 32 && ch <= 126)
    {
      char ch_str[2];
      ch_str[0] = ch;
      ch_str[1] = 0;

      op.cformat("'" + std::string(ch_str) + "'");
    }
```

**Pull request merged**